### ISSUE TITLE
Interrupted I/O

### ISSUE THREAD

**Issue by OmeletWithoutEgg** [NTU-SP/SP-Hw1-release#14]
In non-blocking mode, `write(fd, buf, len)` may return a positive integer less than `len`,
    which means we only sent partial data of `buf` but not all, and we need to send the rest data the next time.

For example, now I want to send `10` bytes of data to the client, but `write` returns `5`, which means it only write `5` bytes to the file descriptor.

Do we need to handle such cases?

**Comment by imingasco** [comment]
Of course. In your example, there's no way to obtain last 5 bytes unless the server sends them properly.

**Comment by 2011eric** [comment]
Is there a way to simulate this kind of situation?

**Comment by imingasco** [comment]
Do you mean to simulate that return value of `write` is less than `len`?

**Comment by 2011eric** [comment]
> Do you mean to simulate that return value of `write` is less than `len`?

yes, that's what I mean

**Comment by imingasco** [comment]
The only way I could come up with to fill up client socket's buffer until it has little space.
